unit_testing/test_labels.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+# pytest unit tests for spinalcordtoolbox.labels
+
+import logging
+from time import time
+
+import numpy as np
+import pytest
+
+import spinalcordtoolbox.labels as sct_labels
+from spinalcordtoolbox.image import Image, zeros_like
+from spinalcordtoolbox.utils import sct_test_path
+from spinalcordtoolbox.types import Coordinate
+from test_image import fake_3dimage, fake_3dimage2
+
+logger = logging.getLogger(__name__)
+
+seg_img = Image(sct_test_path('t2', 't2_seg-manual.nii.gz'))
+t2_img = Image(sct_test_path('t2', 't2.nii.gz'))
+labels_img = Image(sct_test_path('t2', 'labels.nii.gz'))
+
+
+# TODO [AJ] investigate how to parametrize fixtures from test_image.py
+# without redefining the function here
+def fake_3dimage_sct2():
+    """
+    :return: an Image (3D) in RAS+ (aka SCT LPI) space
+    shape = (1,2,3)
+    """
+    i = fake_3dimage2()
+    img = Image(i.get_data(), hdr=i.header,
+                orientation="RPI",
+                dim=i.header.get_data_shape(),
+                )
+    return img
+
+
+def fake_3dimage_sct():
+    """
+    :return: an Image (3D) in RAS+ (aka SCT LPI) space
+    shape = (7,8,9)
+    """
+    i = fake_3dimage()
+    img = Image(i.get_data(), hdr=i.header,
+                orientation="LPI",
+                dim=i.header.get_data_shape(),
+                )
+    return img
+
+
+test_images = [fake_3dimage_sct(), fake_3dimage_sct2(), t2_img]
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_create_labels_empty(test_image):
+    a = test_image.copy()
+    expected = zeros_like(a)
+
+    labels = [Coordinate(l) for l in [[0, 0, 0, 7], [0, 1, 2, 5]]]
+    expected.data[0, 0, 0] = 7
+    expected.data[0, 1, 2] = 5
+
+    b = sct_labels.create_labels_empty(a, labels)
+
+    diff = b.data == expected.data
+    assert diff.all()
```

Comment:
Just curious, why this form instead of using [`numpy.testing`](https://numpy.org/doc/stable/reference/routines.testing.html)?

Reply:
Ah, sorry, my apologies. Disregard this, I misremembered something.